src/dataflow-types/src/plan/join/linear_join.rs

```diff
@@ -43,12 +51,60 @@ pub struct LinearJoinPlan {
     pub final_closure: Option<JoinClosure>,
 }
 
+impl TryFrom<ProtoLinearJoinPlan> for LinearJoinPlan {
+    type Error = TryFromProtoError;
+
+    fn try_from(x: ProtoLinearJoinPlan) -> Result<Self, Self::Error> {
+        let sk: Option<Result<_, _>> = x.source_key.map(TryFrom::try_from);
```

Comment:
shouldn't converting the source_key work just like initial_closure, i.e. 
```
x.source_key.map(|x| x.try_into()).transpose()?
```

Reply:
i'll fix it